### DISCUSSION TITLE
Duplicate data records

### DISCUSSION THREAD

**Discussion by dmAssecor** [statamic-rad-pack/runway#346] [Feature Requests]
Hello, we would like to have a feature where we can duplicate our already created data records with a click of a button in the table. Something like this already exists directly in the Statamic tables. 

![image](https://github.com/duncanmcclean/runway/assets/140411996/c17de48a-8980-4374-ae4f-413d274d6a02)

Would it be possible to implement this?

**Comment by duncanmcclean** [comment]
I'm going to hold off adding a "Duplicate" action into Runway for the minute since it'll need to be configurable (not everyone will want it) but I'll keep this discussion open.

In the meantime, drop this to your `app/Actions` directory:

```php
<?php

namespace App\Actions;

use Statamic\Actions\Action;

class DuplicateRecord extends Action
{
    protected static $title = 'Duplicate';

    public function visibleTo($item)
    {
        return $item instanceof \App\Models\Product;
    }

    public function run($items, $values)
    {
        $items->each(function ($item) {
            $item->replicate()->save();
        });
    }
}
```

You'll need to replace the `$item instanceof ...` bit so it shows up for your Eloquent model(s).

**Reply by dmAssecor** [reply]
Thank you very much. This works great. Exactly what we were looking for. 👍

**Comment by duncanmcclean** [comment]
Runway v6 will add a "Duplicate" action to models.
